src/App.js

```diff
@@ -0,0 +1,178 @@
+  }, []); // run once after mounted , mounted is different from re-rendered
+
+  useEffect(() => {
+    popularMovies();
+  }, [genres]);
+
+  //Input range slider
+
+  let searchByRate = (value) => {
+    let rounded = Math.round(value * 10) / 10
+
+    console.log("value", rounded)
+    setRate(value)
+
+    let filteredData = movieList.filter((movie) => movie.vote_average >= value);
+    setMovies(filteredData);
+  }
+
+  console.log(movies)
+  // Loading message
+  if (movies == null) {
+    return (
+      <Spinner animation="border" role="status">
+        <span className="sr-only">Loading...</span>
+      </Spinner>
+    )
+  }
+  //Navbar: Search input HI i'M IN :.))don't break my shit :D OK WILL DO you can just watch
+  let searchByKeyword = (e) => {
+    keyword = e.target.value;
+    if (keyword === '') {
+      setMovies(movieList)
+    } else {
+      setMovies(movieList.filter((movie) => movie.title.toLowerCase().includes(keyword.toLowerCase())));
+    }
+  }
+  //Navbar: Filter Movies by Genre
+  let filteredMovies = (g) => {
+    if (g === null) {
+      setMovies(movieList)
+    } else {
+      setMovies(movieList.filter((movie) => movie.genre_ids.includes(g)));
+    }
+  }
+  //Navbar: Sort by
+  let sortByPopularity = () => {
+    let sortedMovie = [...movies].sort((a, b) => b.popularity - a.popularity);
+    setMovies(sortedMovie);
+  };
+  let sortByHighestRating = () => {
+    let sortedHighestRating = [...movies].sort((a, b) => b.vote_average - a.vote_average);
+    setMovies(sortedHighestRating);
+  };
+
+  const openModal = async (movieId) => {
+    let url = `https://api.themoviedb.org/3/movie/${movieId}/videos?api_key=${apikey}&language=en-US`;
+    let response = await fetch(url);
+    let data = await response.json();
+    console.log('hehe data video:', data)
+    setTrailer(data.results[0].key)
+    SetModal(true)
+  }
+
+  return (
+    //order of elements on the main page
+    <div id="main" >
+      <Jumbotron style={{ backgroundImage: `url(${bgimage})`, backgroundSize: 'cover' }}>
+      </Jumbotron>
+      <Navbar id="nav" bg="dark" variant="dark" expand="sm" navbar-default>
+        <img src={logo} width="100px" className="d-inline-block align-top" />
+        <FormControl id="searchBox" style={{ maxWidth: "30%" }} Ztype="text" placeholder="Search for movies | serials | actors..." className="mr-sm-2" onChange={(e) => searchByKeyword(e)} />
+        <InputRange
+          maxValue={10}
+          step={0.1}
+          minValue={0}
+          value={rate}
+          onChange={value => searchByRate(value)} />
+
+        {/* step={2}
+  value={this.state.value}
+  onChange={value => this.setState({ value })} /> */}
+
+        <Nav className="mr-auto">
+          <NavDropdown className="text-info dropdown" style={{ color: "#17a2b8" }} title="Genre" id="basic-nav-dropdown dropdown">
+            <NavDropdown.Item className="text-info" onClick={() => filteredMovies(28)} href="#action/3.1">Action</NavDropdown.Item>
```

Comment:
I understand what you're doing here, and two major suggestions:

1. You can make this its own component to at least try and hide this monstrosity in another file
2. You can probably do something like...

at the top of the file
```
const NAV_GENRES = [ { id: 28, name: 'Action'}, { id: 12, name: 'Adventure'} ... ]
```

and here, then do a map
{ NAV_GENRES.map ( g => <NavDrop... onClick = filteredMovies(g.id) ....  }